File: 2.py
s = [1, 2]
sum = 0

n = 2
while s[n-1] < 4_000_000:
    s.append(s[n-1] + s[n-2])
    n += 1

m = 1
while True:
    sum += s[m]
    
    if s[m+1] > 4_000_000:
        break
    
    m += 3
    
print(sum)

File: 2.c
#include <stdio.h>

int main(void) {
    int sum = 0;

    int x = 1;
    int y = 2;

    while (y <= 4000000) {
        sum += y;

        int tempx = x;
        int tempy = y;

        x = tempx + 2*tempy;
        y = 2*tempx + 3*tempy;
    }

    printf("%d\n", sum);

    return 0;
}